src/main/java/org/camunda/community/benchmarks/ProcessDeployer.java

```diff
@@ -43,29 +52,98 @@ public void autoDeploy() {
-            return is;
-        }
-
-        // Replace job types or BPMN id on-the-fly
-
         byte[] stringBytes = is.readAllBytes();
         String fileContent = new String(stringBytes);
-
-        if (config.getJobTypesToReplace()!=null) {
-            // Split by "," if there are multiple task types to be replaced
-            String[] tasksToReplace = {config.getJobTypesToReplace()};
-            if (config.getJobTypesToReplace().contains(",")) {
-                tasksToReplace = config.getJobTypesToReplace().split(",");
+        
+        // First, inject job types for service tasks that don't have them
+        try {
+            fileContent = injectUniqueJobTypes(fileContent);
+        } catch (Exception e) {
+            LOG.warn("Failed to inject job types, proceeding with original content: " + e.getMessage());
+        }
+        
+        // Then apply existing configuration-based replacements
+        if (config.getJobTypesToReplace() != null || config.getBpmnProcessIdToReplace() != null) {
+            if (config.getJobTypesToReplace()!=null) {
+                // Split by "," if there are multiple task types to be replaced
+                String[] tasksToReplace = {config.getJobTypesToReplace()};
+                if (config.getJobTypesToReplace().contains(",")) {
+                    tasksToReplace = config.getJobTypesToReplace().split(",");
+                }
+                for (String taskToReplace: tasksToReplace) {
+                    fileContent = fileContent.replaceAll(taskToReplace, config.getJobType());
+                }
             }
-            for (String taskToReplace: tasksToReplace) {
-                fileContent = fileContent.replaceAll(taskToReplace, config.getJobType());
+            if (config.getBpmnProcessIdToReplace()!=null) {
+                fileContent = fileContent.replaceAll(config.getBpmnProcessIdToReplace(), config.getBpmnProcessId());
             }
         }
-        if (config.getBpmnProcessIdToReplace()!=null) {
-            fileContent = fileContent.replaceAll(config.getBpmnProcessIdToReplace(), config.getBpmnProcessId());
-        }
 
         return new ByteArrayInputStream(fileContent.getBytes());
     }
+
+    /**
+     * Inject unique job types for service tasks that don't have zeebe:taskDefinition
+     * Uses Zeebe's BPMN model API for robust and type-safe BPMN manipulation
+     */
+    private String injectUniqueJobTypes(String bpmnContent) throws Exception {
+        // First, check if zeebe namespace needs to be added
+        String modifiedContent = bpmnContent;
+        if (!bpmnContent.contains("xmlns:zeebe=")) {
+            // Add zeebe namespace declaration using string replacement for safety
+            modifiedContent = bpmnContent.replace(
+                "<bpmn:definitions",
+                "<bpmn:definitions xmlns:zeebe=\"http://camunda.org/schema/zeebe/1.0\""
+            );
+        }
```

Comment:
To robustly check for the presence of the Zeebe namespace, you should look for any xmlns:* attribute that maps to the Zeebe URI, not just xmlns:zeebe=.

Example improvement:
```java
if (!bpmnContent.matches(".*xmlns:\\w+=\"http://camunda.org/schema/zeebe/1.0\".*")) {
    // inject namespace with your preferred prefix (e.g., zeebe)
}
```